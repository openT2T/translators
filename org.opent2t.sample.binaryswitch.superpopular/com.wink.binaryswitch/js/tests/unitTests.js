var test = require('ava');
var translatorPath = require('path').join(__dirname, '..');
var runBinarySwitchTests = require('opent2t-device-binaryswitch/binarySwitchTests');
var deviceData = require('./devicedata');
var OpenT2TLogger = require('opent2t').Logger;
var logger = new OpenT2TLogger("info");
var MockHub = require('opent2t-device-winkhub/mockWinkHub');
var mockHub = new MockHub(logger, deviceData);

var settings = {
    logger,
    test,
    createTranslator: mockHub.createTranslator(translatorPath, deviceData.base_state.data.object_id, deviceData.base_state.data.uuid),
    setTestData: mockHub.setTestData
};

// Run standard binary switch unit tests
runBinarySwitchTests(settings);