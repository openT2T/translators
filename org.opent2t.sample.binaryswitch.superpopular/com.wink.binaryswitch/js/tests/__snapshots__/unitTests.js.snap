exports[`GetPlatform 1`] = `"{\"opent2t\":{\"schema\":\"org.opent2t.sample.binaryswitch.superpopular\",\"translator\":\"opent2t-translator-com-wink-binaryswitch\",\"controlId\":\"F85B0738-6EC0-4A8B-A95A-503B6F2CA0D8\",\"uuid\":\"F85B0738-6EC0-4A8B-A95A-503B6F2CA0D8\"},\"availability\":\"online\",\"pi\":\"F85B0738-6EC0-4A8B-A95A-503B6F2CA0D8\",\"mnmn\":\"Wink\",\"mnmo\":\"Binary Switch (Generic)\",\"n\":\"Mock Name\",\"rt\":[\"org.opent2t.sample.binaryswitch.superpopular\"],\"entities\":[{\"n\":\"Mock Name\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"rt\":[\"oic.d.smartplug\"],\"di\":\"dd6d61bb-5b0b-090b-86ce-c4de9fe59d11\",\"resources\":[{\"href\":\"/power\",\"rt\":[\"oic.r.switch.binary\"],\"if\":[\"oic.if.a\",\"oic.if.baseline\"]}]}]}"`;

exports[`GetPlatformExpanded 1`] = `"{"opent2t":{"schema":"org.opent2t.sample.binaryswitch.superpopular","translator":"opent2t-translator-com-wink-binaryswitch","controlId":"F85B0738-6EC0-4A8B-A95A-503B6F2CA0D8","uuid":"F85B0738-6EC0-4A8B-A95A-503B6F2CA0D8"},"availability":"online","pi":"F85B0738-6EC0-4A8B-A95A-503B6F2CA0D8","mnmn":"Wink","mnmo":"Binary Switch (Generic)","n":"Mock Name","rt":["org.opent2t.sample.binaryswitch.superpopular"],"entities":[{"n":"Mock Name","icv":"core.1.1.0","dmv":"res.1.1.0","rt":["oic.d.smartplug"],"di":"dd6d61bb-5b0b-090b-86ce-c4de9fe59d11","resources":[{"href":"/power","rt":["oic.r.switch.binary"],"if":["oic.if.a","oic.if.baseline"],"id":"power","value":false}]}]}"`;
